examples/search_bar/main.rs

```diff
@@ -0,0 +1,66 @@
+use glib::clone;
+use gtk::glib;
+use gtk::prelude::*;
+
+fn main() {
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.search_bar"),
+        Default::default(),
+    );
+    application.connect_activate(build_ui);
+    application.run();
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+    window.set_default_size(400, 400);
+    window.set_title(Some("Search Bar"));
+
+    let header_bar = gtk::HeaderBar::new();
+    window.set_titlebar(Some(&header_bar));
+
+    let search_button = gtk::ToggleButton::new();
+    search_button.set_icon_name("system-search-symbolic");
+    header_bar.pack_end(&search_button);
+
+    let container = gtk::Box::new(gtk::Orientation::Vertical, 6);
+    window.set_child(Some(&container));
+
+    let search_bar = gtk::SearchBar::new();
+    container.append(&search_bar);
+    search_bar.set_valign(gtk::Align::Start);
+    search_bar.set_key_capture_widget(Some(&window));
+    search_button
+        .bind_property("active", &search_bar, "search-mode-enabled")
+        .flags(glib::BindingFlags::SYNC_CREATE | glib::BindingFlags::BIDIRECTIONAL)
+        .build();
+
+    let entry = gtk::SearchEntry::new();
+    entry.set_hexpand(true);
+    search_bar.set_child(Some(&entry));
+
+    let label = gtk::Label::new(Some("Type to start search"));
+    label.set_vexpand(true);
+    label.set_halign(gtk::Align::Center);
+    label.set_valign(gtk::Align::Center);
+    label.add_css_class("large-title");
+    container.append(&label);
+
+    entry.connect_search_started(clone!(@weak search_button => move |_| {
+        search_button.set_active(true);
+    }));
+
+    entry.connect_stop_search(clone!(@weak search_button => move |_| {
+        search_button.set_active(false);
+    }));
+
+    entry.connect_search_changed(move |entry| {
```

Comment:
OOC, why do you move a strong reference to the entry into this closure?

Reply:
I just thought that's what I should do, if not I can change it.

Reply:
It's not the entry that's probelmatic you're not moving it anywhere. It's the label that you should move a weak reference of it